### ISSUE TITLE
Screening questions page

### ISSUE THREAD

**Issue by brindasasi** [GCTC-NTGC/gc-digital-talent#6068]

**Comment by vd1992** [comment]
Assuming this was an accidental removal of estimate when moving to Product Backlog, restoring the estimate of 3

**Comment by petertgiles** [comment]
> Should the label for the `textarea` be the question rather than "Your answer to question {number}"? We will need to associate the question to the input for AT's anyway and the best way to do that is with a `label`.

I commented on this in my review.  Unfortunately it doesn't look good for multiline questions - of which I suspect will be most of them.

**Comment by esizer** [comment]
> I commented on this in my review. Unfortunately it doesn't look good for multiline questions - of which I suspect will be most of them.

This is going to be a very large issue since AT users typically navigate forms by tabbing through each input individually and hear the label read out to them. So what they will hear is something like:

- Textarea, Answer to question 1
- Textarea, Answer to question 2

With no concept of what the actual question is. So they would then need to go back up to hear the question be read and make an assumption that what they just heard was question 1 with no certainty unless they go back up and hear the heading being read. That means, in order to reliably answer a question the user would need to do:

- `tab`
- `AT Key + up`
- `AT Key + up`
- `tab`

Hearing the text in the opposite order it was meant to be read. Which would get old, very quick 😢

**Comment by substrae** [comment]
Is it not reasonable to wrap the `textarea` and question inside of a `fieldset` and use the `legend` to describe the question?

**Comment by esizer** [comment]
> Is it not reasonable to wrap the `textarea` and question inside of a `fieldset` and use the `legend` to describe the question?

No, since the [legend is not read out on all AT](https://www.w3.org/WAI/tutorials/forms/grouping/#:~:text=Note%3A%20Depending%20on%20the%20configuration%2C%20some%20screen%20readers%20read%20out%20the%20legend%20either%20with%20every%20form%20element%2C%20once%2C%20or%2C%20rarely%2C%20not%20at%20all.). 😢 Also, it is [required to make the labels self-explanatory](https://www.w3.org/WAI/tutorials/forms/grouping/#:~:text=Make%20the%20individual%20labels%20sufficiently%20self%2Dexplanatory%20for%20situations%20in%20which%20legends%20are%20not%20read%20aloud%2C%20without%20repeating%20the%20legend%20in%20every%20label.) and "Answer to question 1" is not self-explanatory without knowing what the question is, especially with common navigation technique of [listing form inputs](https://www.freedomscientific.com/training/jaws/hotkeys/#:~:text=NUM%20PAD%20PLUS-,List%20Form%20Fields,-INSERT%2BF5) to quickly navigate a form. 

When listing form inputs a user would just hear:

- Answer to question 1
- Answer to question 2

With now knowledge of what the actual questions are.

Fieldsets are meant for grouping multiple similar inputs and typically not used for single inputs.

**Comment by substrae** [comment]
ATs not reading out `legends` seems like an AT problem to me, not an incorrect use of the HTML 🤔 

Another option is to use a more descriptive `labelled-by`... But at any rate, this is a pretty common pattern used out in the wild, where the question isn't the label directly. Making the question the label seems like a bad option all around, both because it's not demanding an action and because the length would exceed a reasonable label format.

**Comment by esizer** [comment]
> Making the question the label seems like a bad option all around, both because it's not demanding an action and because the length would exceed a reasonable label format.

I would argue that asking a question, associating it with an input and making it required is demanding an action (response to the question). Though, I agree it is very ugly and we should avoid it if possible 😅 

> The link you sent has an instance of this happening right in their examples:

Yes, that is why I specified "... **_typically_** not used for single inputs". It can be done, but since the legend is not guaranteed to be read out to ATs, it is not the best pattern available and it should be treated as supplementary information, not key information. 

> Another option is to use a more descriptive `labelled-by`

I think this may be a good solution, `labelled-by` [supports multiple IDs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby#values) and we are already doing this with the `aria-describedby` 👍 . We will just need to update our inputs to accept additional values here and test it. I believe the resulting HTML should look similar to the following:

```html
<h2>Question 1</h2>
<p id="question-1">What is your name?</p>
<label id="question-1-label" for="question-1-answer">Answer to question 1</label>
<textarea id="question-1-answer" label="Answer to question 1" aria-labelledby="question-1 question-1-label">Answer</textarea>
```

**Comment by substrae** [comment]
Hmm, I do still take issue with: 

>Yes, that is why I specified "... typically not used for single inputs". It can be done, but since the legend is not guaranteed to be read out to ATs, it is not the best pattern available and it should be treated as supplementary information, not key information.

If the WCAG website has an example of exactly this behaviour in the context of using the `legend` element, I'm inclined to suggest we're okay to use it. If we were to decide not to, it basically removes the purpose of the `legend` element entirely, and that impacts not only this element, but all fieldsets, including the new design for our dates. Each input inside that fieldset would no longer have the context of the larger `legend` to define what the `day`, `month`, and `year` apply to.
